### DISCUSSION TITLE
Change the PK value while keeping the FK links

### DISCUSSION THREAD

**Discussion by CharlotteJu** [simolus3/drift#2893] [Q&A] [answered]
Hi Simon ! 
In my application I create temporary PK(id) (waiting for the server to return the final value).
In Drift, is there a way to change the PK value while retaining the FK links of the other tables (when there is the OnUpdate Cascade)?
In SQL this would be equivalent to:
UPDATE table SET Id = 1 WHERE Id = 4728

**Comment by simolus3** [comment, marked as answer]
Hi! Sure, you can construct that update statement in drift with

```dart
  Future<void> updateId(int oldId, int newId) async {
    final stmt = update(table)..where((row) => row.id.equals(oldId));
    await stmt.write(TableCompanion(id: Value(newId)));
  }
```

Is this what you were looking for? As you've said, this should also update foreign keys if they have an `ON UPDATE CASCADE` option.

**Comment by CharlotteJu** [comment]
Thank's, it's perfect !
